src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorProjectChangePublisher.cs

```diff
@@ -217,7 +217,7 @@ protected virtual void SerializeToFile(ProjectSnapshot projectSnapshot, string p
 
             if (tempFileInfo.Exists)
             {
-                Debug.Fail($"'{tempFileInfo.FullName}' should not exist but it does. This could be caused by failures during serialization or early process termination.");
```

Comment:
As the person guilty of asking for this `Debug.Fail`, I have to say that the main reason for this was to figure out why this happens and make sure we don't even run into those inconsistent states in the first place. Without this, we'll just never know if we are continuously running into this state. Is your point that this isn't useful because we'll never be able to prevent this inconsistency from happening?

Reply:
Yup! If the process terminates or you get any sort of access restriction exceptions when moving/writing the file it'll stick around. Sadly any sort of file manipulation is not guarunteed to have an exact result every time.